packages/devtools_app/test/association_variable_test.dart

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+import 'package:vm_service/vm_service.dart';
+
+import 'package:devtools_app/src/debugger/debugger_controller.dart';
+import 'package:devtools_app/src/debugger/debugger_model.dart';
+import 'package:devtools_app/src/globals.dart';
+import 'package:devtools_app/src/service_manager.dart';
+import 'package:devtools_app/src/vm_service_wrapper.dart';
+
+void main() {
+  ServiceConnectionManager manager;
+  DebuggerController debuggerController;
+
+  setUp(() {
+    manager = ServiceConnectionManager();
+    manager.service = MockVmServiceWrapper();
+    when(manager.service.onDebugEvent).thenAnswer((_) {
+      return const Stream.empty();
+    });
+    when(manager.service.onIsolateEvent).thenAnswer((_) {
+      return const Stream.empty();
+    });
+    when(manager.service.onStdoutEvent).thenAnswer((_) {
+      return const Stream.empty();
+    });
+    when(manager.service.onStderrEvent).thenAnswer((_) {
+      return const Stream.empty();
+    });
+    globals[ServiceConnectionManager] = manager;
+    debuggerController = DebuggerController();
+    debuggerController.isolateRef =
+        IsolateRef(id: '1', number: '2', name: 'main');
+  });
+
+  tearDown(() {
+    globals[ServiceConnectionManager] = null;
+  });
```

Comment:
is this necessary if the global  var is set in each setUp call?

Reply:
No, good point - removed.